src/core/MOM_forcing_type.F90

```diff
@@ -3539,17 +3607,21 @@ subroutine rotate_forcing(fluxes_in, fluxes, turns)
   type(forcing), intent(inout) :: fluxes      !< Rotated forcing structure
   integer, intent(in) :: turns                !< Number of quarter turns
 
-  logical :: do_ustar, do_water, do_heat, do_salt, do_press, do_shelf, &
+  logical :: do_ustar, do_taumag, do_water, do_heat, do_salt, do_press, do_shelf, &
       do_iceberg, do_heat_added, do_buoy
 
-  call get_forcing_groups(fluxes_in, do_water, do_heat, do_ustar, do_press, &
+  call get_forcing_groups(fluxes_in, do_water, do_heat, do_ustar, do_taumag, do_press, &
       do_shelf, do_iceberg, do_salt, do_heat_added, do_buoy)
 
-  if (do_ustar) then
```

Comment:
Did the `do_ustar` etc flags need to be replaced with `associate()` checks here?  Maybe fixed a bug related to `ustar_gustless`?

Reply:
In this case, the `do_ustar` logicals don't need to be replaced yet, but in a subsequent commit we will be only allocating the components ustar or tau_mag that are actually used, so at that point using this group logical to test one of them will no longer be useful for figuring out whether the other ones can be rotated.  Doing a separate test on each of the component arrays should be both more general and safer.